### ISSUE TITLE
Xenon node state doesn't match primary miner chainstate

### ISSUE THREAD

**Issue by diwakergupta** [blockstack/stacks-blockchain#2121]
I spun up a Xenon miner with local bitcoind in the past hour.

Currently Xenon miner's `v2/info`:
```json
{
  "peer_version": 385875968,
  "pox_consensus": "87fe48fd72d55157fc1c6b450bcb5fe2a3913140",
  "burn_block_height": 1895953,
  "stable_pox_consensus": "87ac064b7c507fbf5972e893f3b67b54c425bdc0",
  "stable_burn_block_height": 1895946,
  "server_version": "blockstack-core 0.0.1 => 23.0.0.0 (fix/xenon-fixes:679d2ef+, release build, linux [x86_64])",
  "network_id": 2147483648,
  "parent_network_id": 118034699,
  "stacks_tip_height": 866,
  "stacks_tip": "aaf0dfca8ead5817d72c1d8ee9ed824c29680f8c47491325aa018aeb23380fe4",
  "stacks_tip_consensus_hash": "124444aa86ba6fa5621dfdfc80067fb0f5b4e71e",
  "unanchored_tip": "03508074f60c3d4c12d005755669f1d49223f16fe8242940127d5023207ca936",
  "exit_at_block_height": null
}
```

Local miner's `v2/info`:
```json
{
  "peer_version": 385875968,
  "pox_consensus": "87fe48fd72d55157fc1c6b450bcb5fe2a3913140",
  "burn_block_height": 1895953,
  "stable_pox_consensus": "87ac064b7c507fbf5972e893f3b67b54c425bdc0",
  "stable_burn_block_height": 1895946,
  "server_version": "blockstack-core 0.0.1 => 23.0.0.0 (main:57953c0ac, debug build, macos [x86_64])",
  "network_id": 2147483648,
  "parent_network_id": 118034699,
  "stacks_tip_height": 616,
  "stacks_tip": "da211fd1f399ae5cf3f4de930a0d9ff5ed23a276aabab074b3b315db99d52052",
  "stacks_tip_consensus_hash": "6ae829ce436a9400a8982d7ff1a78da90b342053",
  "unanchored_tip": "a1504f8ca488afc4e1df8f73ed3a9bec70e32fab968bc642e24d466819dd582a",
  "exit_at_block_height": null
}
```

Notice that `burn_block_height` and pox-consensus hashes match, but `stacks_tip_height` (and corresponding consensus hash) does not.

In my node logs, I see a lot of warnings like:
```
WARN [1606940365.095017] [src/chainstate/stacks/block.rs:324] [ThreadId(4)] Failed to verify proof '7d3675cfad30abec894df7636f3a7a7c6ff52f586f136812a60a2e96f15ee2c1acaf65d47139463487b403e58551624c0a7e860a5d0614c23e3e8c4caeb49ebbf539f4abcca12abd33e4bb2f8854cb04'
WARN [1606940365.095167] [src/chainstate/stacks/block.rs:340] [ThreadId(4)] Invalid Stacks block header 8c6632a9c56a77cb58e0f261493c22a9401fd2e08f6925eb43257a91009c0fd0: leader VRF key 6e0283846fd0923d2cd0bd09cae9418973ab3e801502a1060a6c8a14ac175708 did not produce a valid proof over e915ba2ccf155a18799e1e8a7b01c9d86233a0f71d8329396c35148f3b6918a9
WARN [1606940365.095201] [src/chainstate/stacks/db/blocks.rs:2936] [ThreadId(4)] Invalid block, could not validate on burnchain: e001496680771636ea5bb31944c62ca32e02ee0c/8c6632a9c56a77cb58e0f261493c22a9401fd2e08f6925eb43257a91009c0fd0
WARN [1606940365.095351] [src/chainstate/stacks/db/blocks.rs:3054] [ThreadId(4)] Invalid block 8c6632a9c56a77cb58e0f261493c22a9401fd2e08f6925eb43257a91009c0fd0: does not correspond to burn chain state
```

Full debug logs are attached.
[xenon-miner.log.gz](https://github.com/blockstack/stacks-blockchain/files/5631838/xenon-miner.log.gz)

**Comment by friedger** [comment]
Is it eligible for a bug bounty?

**Comment by jcnelson** [comment]
> In my node logs, I see a lot of warnings like:

Thanks for the logs @diwakergupta!  My diagnosis is that your node is talking to a Stacks node who has a different view of the Bitcoin testnet.  I notice that you're running your own testnet Bitcoin node locally, and I notice that a particular peer (34.86.161.84) keeps serving your node blocks that it cannot process.  Specifically, this peer and _only_ this peer is responsible for serving you _all_ of the blocks that trigger this error.  Your node is correctly deducing that this remote peer is behaving badly, and temporarily banning it.

**Comment by diwakergupta** [comment]
Thanks @jcnelson ! As @psq pointed out on Discord, `34.86.161.84` is the Xenon miner, so this would suggest the bitcoin node Xenon miner is talking to has a different view of the burnchain than my local bitcoind.

cc/ @wileyj @CharlieC3 let's follow up offline on diagnosing this further. I should have logs / state from my local bitcoind (though it's not currently running)

**Comment by diwakergupta** [comment]
I'm able to reproduce this problem with a follower, pointed to Xenon's bitcoind. **That strongly suggests the problem lies with the bitcoind that Xenon is running**, though I'm surprised why a miner with it's own local bitcoind is unable to work around whatever the problem is.

Follower's `v2/info`:

```json
{
  "peer_version": 385875968,
  "pox_consensus": "6edc22013cb5d6031af5992daac695c9509b3fe0",
  "burn_block_height": 1896127,
  "stable_pox_consensus": "56ce2506572068bcd78d7071c4b1d10cb5398973",
  "stable_burn_block_height": 1896120,
  "server_version": "blockstack-core 0.0.1 => 23.0.0.0 (main:c176ddabf, debug build, macos [x86_64])",
  "network_id": 2147483648,
  "parent_network_id": 118034699,
  "stacks_tip_height": 616,
  "stacks_tip": "da211fd1f399ae5cf3f4de930a0d9ff5ed23a276aabab074b3b315db99d52052",
  "stacks_tip_consensus_hash": "6ae829ce436a9400a8982d7ff1a78da90b342053",
  "unanchored_tip": "a1504f8ca488afc4e1df8f73ed3a9bec70e32fab968bc642e24d466819dd582a",
  "exit_at_block_height": null
}
```

Xenon miner's `v2/info`:
```json
{
  "peer_version": 385875968,
  "pox_consensus": "6edc22013cb5d6031af5992daac695c9509b3fe0",
  "burn_block_height": 1896127,
  "stable_pox_consensus": "56ce2506572068bcd78d7071c4b1d10cb5398973",
  "stable_burn_block_height": 1896120,
  "server_version": "blockstack-core 0.0.1 => 23.0.0.0 (fix/xenon-fixes:679d2ef+, release build, linux [x86_64])",
  "network_id": 2147483648,
  "parent_network_id": 118034699,
  "stacks_tip_height": 1014,
  "stacks_tip": "61a865781fe0669b55e9517879e48b86131de7ffe2a6d2c175171663d0683946",
  "stacks_tip_consensus_hash": "6edc22013cb5d6031af5992daac695c9509b3fe0",
  "unanchored_tip": "e350aa658c3a7a893f7bb8cedda27deaefc037dd62eca36cb07d7b4ef73344b6",
  "exit_at_block_height": null
}
```

Debug logs attached: [xenon-follower-diwaker.log.gz](https://github.com/blockstack/stacks-blockchain/files/5643693/xenon-follower-diwaker.log.gz)

**Comment by diwakergupta** [comment]
Should have been addressed, I'll reopen if it happens again.
